### PR TITLE
tests: add stage snaps integration test (CRAFT-68)

### PR DESCRIPTION
**Pull request by cmatsuoka**
Verify if stage-snaps are correctly unpacked into the part install dir.

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: tests/integration/lifecycle/test_stage_snaps.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        ctx.execute(actions[0])
+
+    snaps = list(Path("parts/foo/stage_snaps").glob("*.snap"))
+    assert len(snaps) == 1
+    assert snaps[0].name == "test-snap.snap"
+
+    ctx.execute(actions[1])
+
+    foo_install_dir = Path(new_dir / "parts" / "foo" / "install")
+    assert (foo_install_dir / "meta.basic" / "snap.yaml").is_file()
```
Comment (sergiusens) [inline tests/integration/lifecycle/test_stage_snaps.py:64]: the file is test-snap and the name in snap.yaml is probably basic. For clarity, I would either:

- clarify
- rename the snap name
- rename the snap file

File: tests/integration/lifecycle/test_stage_snaps.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+
+def test_stage_snap_error(new_dir, fake_snap_command):
```
Comment (sergiusens) [inline tests/integration/lifecycle/test_stage_snaps.py:67]: can you clarify the type of error? Is this download error, unpack error? Do we need to have one for each of these?

**Review by sergiusens — APPROVED** [review]
Looks good, approved with comments.

**Review by facundobatista — APPROVED** [review]
Like it.

File: tests/integration/lifecycle/test_stage_snaps.py (hunk 1 above)
Comment (cmatsuoka) [reply tests/integration/lifecycle/test_stage_snaps.py:64]: It bothered me too, I'll rename the snap file to be consistent.

File: tests/integration/lifecycle/test_stage_snaps.py (hunk 2 above)
Comment (cmatsuoka) [reply tests/integration/lifecycle/test_stage_snaps.py:67]: Added separate tests for download and pull error.

**Pull request merged**